### DISCUSSION TITLE
missing token header in multiplayer room

### DISCUSSION THREAD

**Discussion by t1ww** [ppy/osu#27020] [Q&A]
i when start playing in multi, it shows notification "you are not able to submit score, missing token header" and it abort my gameplay
not sure what it means but this affects to just me, other player in the room can continue to play and doesnt have the notification
![image](https://github.com/ppy/osu/assets/95153968/965c3e22-0544-4da9-bdbf-457c812d4e5a)

**Comment by bdach** [comment]
if this is arch linux, please stop using AUR packages and use the appimage instead

if this is not arch linux, please upload logs

**Reply by t1ww** [reply]
this is window so here's the log
[1707053004.updater.log](https://github.com/ppy/osu/files/14162976/1707053004.updater.log)
[1707053004.runtime.log](https://github.com/ppy/osu/files/14162977/1707053004.runtime.log)
[1707053004.network.log](https://github.com/ppy/osu/files/14162978/1707053004.network.log)
[1707053004.database.log](https://github.com/ppy/osu/files/14162979/1707053004.database.log)
[1707053004.performance.log](https://github.com/ppy/osu/files/14162981/1707053004.performance.log)

**Reply by bdach** [reply]
where did you install the game from?

**Reply by t1ww** [reply]
hi sorry for late reply, i installed from the official osu website back in like 2021 iirc

**Comment by SashaCanfield03** [comment]
Let me know if you find a solution because I have the same problem too. It looks like it causes a problem for score submission in singleplayer too, so I assumed it had something to do with how VersionHash is calculated, but I didn't see any changes to that variable. I also tried packet sniffing with wireshark when running the AppImage to find a hash, but I didn't see one. I'll probably just have to decompile the whole thing and search for differences between the AppImage and source code.

I did find a quick fix though. In osu.Game/Screens/Play/SubmittingPlayer.cs line 168, by setting osu.Game/Screens/Play/SubmittingPlayer.cs to false instead of true, you can still play in multiplayer. It doesn't fix the score submission issue though.

I still think it's worth it to have higher fps and other quality of life enhancements instead of score submission, but it is really annoying, and I hope they fix it soon. Also, I tried the AppImage version in my testing, and the logger said it got the token, but everything under recent plays on the lazer size of my profile was from 12 days ago. It showed up on the osu side of the site though.

**Reply by bdach** [reply]
Please do not do what you're doing and just use the appimages. By messing around with game source you are risking a restriction. There are measures in place to ensure fairness in the releases we publish which will not be present if you build from source.
